include/oboe/AudioStream.h

```diff
@@ -395,6 +395,16 @@ class AudioStream : public AudioStreamBase {
 
     AudioFormat mNativeFormat = AudioFormat::Invalid;
 
+    /**
+     * Update mFramesWritten.
+     */
+    virtual void updateFramesWritten() = 0;
+
+    /**
+     * Update mFramesRead.
+     */
+    virtual void updateFramesRead() = 0;
```

Comment:
As above. Also might be worth moving the 2 member variables below into a protected section as well. The more we can reduce the API surface the less likely we will be to break it.

Reply:
It is already "protected:".